### PR TITLE
remove padding attribute

### PR DESCRIPTION
**Pull request by jmitrevs**
# Description

This PR removes the "padding" attribute from hls4ml's IR. Given that it we specify the actual pad values in the IR, this is redundant, and it is also keras-specific, so it becomes unnatural to set in pytorch and onnx, since you can get paddings that do not map exactly.

## Type of change

- [x] Other (Specify):  remove keras-specific redundant feature

## Tests

This is supposed to produce no changes, so original tests should continue working. That functions as a test.

## Checklist

- [x] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by vloncar** [comment]
I verified that the tests run ok, they failed due to a timeout. We'll address this in a separate PR.

**Review by vloncar — APPROVED** [review]
